packages/financial-templates-lib/src/price-feed/StLouisFedGovPriceFeed.ts

```diff
@@ -68,10 +79,15 @@ export class StLouisFedGovPriceFeed extends PriceFeedInterface {
       lastUpdateTimestamp: this.lastUpdateTime
     });
 
-    this.currentPrice = await this._getHistoricalPrice(currentTime);
-    this.lastUpdateTime = currentTime;
+    if (this.useStLouisLocalTime) {
+      const stLouisTime = this.getTimestampInTimeZone('America/Chicago');  // St Louis uses America/Chicago timezone.
+      this.currentPrice = await this._getHistoricalPrice(stLouisTime);     // use stLouis time to retrieve price.
+    }
+    else {
+      this.currentPrice = await this._getHistoricalPrice(currentTime)
+    }
+    this.lastUpdateTime = currentTime;  // make sure that the last update time always reflects the bot time in any case.
```

Comment:
`this.lastUpdateTime = currentTime;` will always store bot time, which is incorrect. Once we make the above change, it should store either bot time or st louis time, depending on boolean.

Reply:
Using current time for **lastUpdated** ensures accurate calculation of **historicalLookbackWindow**.